repository.go

```diff
@@ -426,15 +467,115 @@ func (r *Repository) BranchingModel(rbmo *RepositoryBranchingModelOptions) (*Bra
 	return decodeBranchingModel(response)
 }
 
-func (r *Repository) buildJsonBody(body map[string]interface{}) string {
+func (r *Repository) ListEnvironments(opt *RepositoryEnvironmentsOptions) (*Environments, error) {
+	urlStr := r.c.requestUrl("/repositories/%s/%s/environments/", opt.Owner, opt.RepoSlug)
+	res, err := r.c.executeRaw("GET", urlStr, "")
+	if err != nil {
+		return nil, err
+	}
 
-	data, err := json.Marshal(body)
+	bodyBytes, err := ioutil.ReadAll(res)
 	if err != nil {
-		pp.Println(err)
-		os.Exit(9)
+		return nil, err
 	}
 
-	return string(data)
+	bodyString := string(bodyBytes)
+	return decodeEnvironments(bodyString)
+}
+
+func (r *Repository) AddEnvironment(opt *RepositoryEnvironmentOptions) (*Environment, error) {
+	body := r.buildEnvironmentBody(opt)
+	urlStr := r.c.requestUrl("/repositories/%s/%s/environments/", opt.Owner, opt.RepoSlug)
+	res, err := r.c.execute("POST", urlStr, body)
+	if err != nil {
+		return nil, err
+	}
+
+	return decodeEnvironment(res)
+}
+
+func (r *Repository) DeleteEnvironment(opt *RepositoryEnvironmentDeleteOptions) (interface{}, error) {
+	urlStr := r.c.requestUrl("/repositories/%s/%s/environments/%s", opt.Owner, opt.RepoSlug, opt.Uuid)
+	return r.c.execute("DELETE", urlStr, "")
+}
+
+func (r *Repository) GetEnvironment(opt *RepositoryEnvironmentOptions) (*Environment, error) {
+	urlStr := r.c.requestUrl("/repositories/%s/%s/environments/%s", opt.Owner, opt.RepoSlug, opt.Uuid)
+	res, err := r.c.execute("GET", urlStr, "")
+	if err != nil {
+		return nil, err
+	}
+
+	return decodeEnvironment(res)
+}
+
+func (r *Repository) ListDeploymentVariables(opt *RepositoryDeploymentVariablesOptions) (*DeploymentVariables, error) {
+	/*
+	 * Deployment variables apparently cannot (yet?) be filtered, paged or sorted.
+	 */
+/*
+	params := url.Values{}
+	if opt.Query != "" {
+		params.Add("q", opt.Query)
+	}
+
+	if opt.Sort != "" {
+		params.Add("sort", opt.Sort)
+	}
+
+	if opt.PageNum > 0 {
+		params.Add("page", strconv.Itoa(opt.PageNum))
+	}
+
+	if opt.Pagelen > 0 {
+		params.Add("pagelen", strconv.Itoa(opt.Pagelen))
+	}
+
+	if opt.MaxDepth > 0 {
+		params.Add("max_depth", strconv.Itoa(opt.MaxDepth))
+	}
+ */
```

Comment:
Yeah, I will open a support ticket with Atlassian to enquire about it.  It *seems* like an oversight, imo, but if they say it was intentionally not added then I will happily create a new PR to remove the entire commented block.

Reply:
I'm discovering lots of little inconsistencies like this in BitBucket's API.  This one was just more glaring because I originally copied & pasted the code from `repository.ListPipelineVariables` and then spent a while scratching my head trying to figure out why this was failing.  Took a bit of trial & error, and wild guessing, on my part before I tried removing the request params and it started working.  🤦

Reply:
Exactly. BitBucket API v2 had been sometime crush or behave strangely. As far as I know, time will solve that.

Reply:
Ok, the commented-out code has been re-enabled.